WalletWasabi/Services/WalletService.cs

```diff
@@ -1068,6 +1074,64 @@ public IEnumerable<string> GetNonSpecialLabels()
 				.Distinct();
 		}
 
+		private async Task RefreshHistoryAsync(SemaphoreSlim semaphore, SmartCoin coin)
```

Comment:
We use AsyncLock.

Reply:
Can I use AsyncLock like this? Semaphore let me have a counter which is decreased if it is used until zero and increased if released.